fossology/fossology
# fix: improve duplicate file warning UI (#2900)

## Description
This pull request Fixes #2900 .  The current popover warning has several issues, such as inconsistent positioning, lack of visibility, and unclear actions for users.

The proposed solution replaces the popover with a centered, responsive modal dialog that:

Clearly displays duplicate files and their details (file name, upload date).
Provides actionable buttons:
"Continue Anyway": Proceeds with the upload.
"Clear Selection": Resets the file selection.
Enhances user experience with better visibility, accessibility, and mobile responsiveness.
## How to Test
Upload a new file: No warning modal should appear.
Upload an existing file: The warning modal should appear with duplicate file details and action buttons.
Test "Continue Anyway" and "Clear Selection" buttons to ensure they work as expected.
Verify responsiveness on different screen sizes (desktop, tablet, mobile).
## Proof of work

<img width="850" alt="image" src="https://github.com/user-attachments/assets/9a1ee15f-ea64-45ee-8078-9cc12daecb51" />
